mapie/conformity_scores/conformity_scores.py

```diff
@@ -0,0 +1,177 @@
+        self,
+        y_pred: ArrayLike,
+        conformity_scores: ArrayLike,
+    ) -> NDArray:
+        """
+        Placeholder for ``get_estimation_distribution``.
+        Subclasses should implement this method!
+
+        Compute samples of the estimation distribution from the predicted
+        values and the conformity scores.
+
+        Parameters
+        ----------
+        y_pred: NDArray
+            Predicted values.
+
+        conformity_scores: NDArray
+            Conformity scores.
+
+        Returns
+        -------
+        NDArray
+            Observed values.
+        """
+
+    def check_consistency(
+        self,
+        y: ArrayLike,
+        y_pred: ArrayLike,
+        conformity_scores: ArrayLike,
+    ) -> None:
+        """
+        Check consistency between the following methods:
+        ``get_estimation_distribution`` and ``get_signed_conformity_scores``
+
+        The following equality should be verified:
+        ``self.get_estimation_distribution(
+            y_pred, self.get_conformity_scores(y, y_pred)
+        ) == y``
+
+        Parameters
+        ----------
+        y: NDArray
+            Observed values.
+
+        y_pred: NDArray
+            Predicted values.
+
+        Raises
+        ------
+        ValueError
+            If the two methods are not consistent.
+        """
+        score_distribution = self.get_estimation_distribution(
+            y_pred, conformity_scores
+        )
+        abs_conformity_scores = np.abs(np.subtract(score_distribution, y))
+        max_conf_score = np.max(abs_conformity_scores)
+        if max_conf_score > self.eps:
+            raise ValueError(
+                "The two functions get_conformity_scores and "
+                "get_estimation_distribution of the ConformityScore class "
+                "are not consistent. "
+                "The following equation must be verified: "
+                "self.get_estimation_distribution(y_pred, "
+                "self.get_conformity_scores(y, y_pred)) == y. "  # noqa: E501
+                f"The maximum conformity score is {max_conf_score}."
+                "The eps attribute may need to be increased if you are "
+                "sure that the two methods are consistent."
+            )
+
+    def get_conformity_scores(
+        self,
+        y: ArrayLike,
+        y_pred: ArrayLike,
+    ) -> NDArray:
+        """
+        Get the conformity score considering the symmetrical property if so.
+
+        Parameters
+        ----------
+        y: NDArray
+            Observed values.
+
+        y_pred: NDArray
+            Predicted values.
+
+        Returns
+        -------
+        NDArray
+            Conformity scores.
+        """
+        conformity_scores = self.get_signed_conformity_scores(y, y_pred)
+        if self.consistency_check:
+            self.check_consistency(y, y_pred, conformity_scores)
+        if self.sym:
+            conformity_scores = np.abs(conformity_scores)
```

Comment:
Why if symmetric, we get the absolute value of the scores, could you just lay out the reasoning for me?